src/components/cart/cartList.js

```diff
@@ -0,0 +1,224 @@
+  const [isQtyUpdated, setQtyUpdate] = useState(true);
+  const totalReducer = () =>
+    itemsInCart.reduce((acc, item) => acc + item.price * item.quantity, 0);
+
+  const CartUpdate = async ({ type, payLoad }) => {
+    const _id = payLoad;
+    const itemFound = itemsInCart.find((item) => item._id === _id);
+    let updatedQuantity;
+    if (type === "INCREMENT") {
+      updatedQuantity = itemFound.quantity + 1;
+    } else if (type === "DECREMENT") {
+      updatedQuantity = itemFound.quantity - 1;
+      // return updatedQuantity === 0 && deleteCartItem(_id);
+      if (updatedQuantity === 0) {
+        return deleteCartItem(_id);
+      }
+    }
+    try {
+      setQtyUpdate(false);
+      const { data } = await axios.post(`${BACKEND_URL}cart/${_id}`, {
+        _id: _id,
+        quantity: updatedQuantity
+      });
+      payLoad = {
+        _id: _id,
+        quantity: updatedQuantity
+      };
+      console.log("passing load is", payLoad);
+      setQtyUpdate(true);
+      if (data.success) {
+        cartDispatch({ type: "UPDATE", payLoad: payLoad });
+      }
+    } catch (err) {
+      console.error("Error Occured", err);
+    }
+    return (
+      <>
+        <DataLoader />
+      </>
+    );
+  };
+
+  const deleteCartItem = async (_id) => {
+    try {
+      console.log("inside delete");
+      const { data } = await axios.delete(`${BACKEND_URL}cart/${_id}`);
+      console.log(data);
+      cartDispatch({ type: "REMOVE", payLoad: _id });
+      if (data.success) {
+        toast.dark("Removed from cart", {
+          position: "bottom-left",
+          autoClose: 3000,
+          hideProgressBar: true
+        });
+      }
+    } catch (err) {
+      console.error("Error happened", err);
+    }
+  };
+
+  const wishlistHandler = async (product) => {
+    console.log("Incoming data", product);
+
+    const itemFound = wishList.find((item) => item._id === product._id);
+    console.log("wish search", itemFound);
+
+    if (itemFound) {
+      try {
+        const { data } = await axios.delete(
+          `${BACKEND_URL}wishlist/${product._id}`
+        );
+        if (data.success) {
+          wishDispatch({ type: "REMOVE", payLoad: product._id });
+          toast.dark("Item removed from wishlist", {
+            position: "bottom-left",
+            autoClose: 3000,
+            hideProgressBar: true
+          });
+        }
+      } catch (err) {
+        console.error(err);
+      }
+    } else {
+      try {
+        const { data } = await axios.post(`${BACKEND_URL}wishlist`, {
+          _id: product._id
+        });
+        // console.log("posted data", data);
+        if (data.success) {
+          wishDispatch({ type: "ADD_TO_WISHLIST", payLoad: product });
+          toast.success("Item added to wish list", {
+            position: "bottom-left",
+            autoClose: 3000,
+            hideProgressBar: true
+          });
+        }
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  };
+

```

Comment:
This function is the same as used in `productList.js`. Try not to repeat yourself